### PR TITLE
MOL-207: Once Click Payments

### PR DESCRIPTION
**Pull request by boxblinkracer**

### CODE REVIEW

File: src/Resources/app/storefront/src/mollie-payments/plugins/creditcard-components-sw64.plugin.js [hunk 1]
```diff
@@ -225,6 +215,25 @@ export default class MollieCreditCardComponentsSw64 extends Plugin {
...
 
+        const mandateId = this.getMandateCheckedValue();
+        // If the mandateId is valid, that means there is a mandate already selected,
+        // so we have to call the API to save it
+        // and then we continue by submitting our original payment form.
+        if (this.isValidSelectedMandate(mandateId)) {
+            this.client.get(
+                me.options.shopUrl + '/mollie/components/store-mandate-id/' + me.options.customerId + '/' + mandateId,
+                () => {
+                    paymentForm.submit();
```
Comment (boxblinkracer) [inline src/Resources/app/storefront/src/mollie-payments/plugins/creditcard-components-sw64.plugin.js:226]: i think we need to call this in here
me.continueShopwareCheckout(paymentForm);

that is the "real" way to continue with the normal process

File: src/Resources/app/storefront/src/mollie-payments/plugins/creditcard-components.plugin.js [hunk 2]
```diff
@@ -222,6 +216,25 @@ export default class MollieCreditCardComponents extends Plugin {
...
         ) {
+            const mandateId = this.getMandateCheckedValue();
+            // If the mandateId is valid, that means there is a mandate already selected,
+            // so we have to call the API to save it
+            // and then we continue by submitting our original payment form.
+            if (this.isValidSelectedMandate(mandateId)) {
+                this.client.get(
+                    me.options.shopUrl + '/mollie/components/store-mandate-id/' + me.options.customerId + '/' + mandateId,
+                    () => {
+                        paymentForm.submit();
```
Comment (boxblinkracer) [inline src/Resources/app/storefront/src/mollie-payments/plugins/creditcard-components.plugin.js:227]: i think also here 
me.continueShopwareCheckout(paymentForm);

File: src/Resources/config/config.xml [hunk 3]
```diff
@@ -77,6 +77,26 @@
             <helpText lang="de-DE">Zeigt Kreditkartenfelder direkt im Shopware shop an. Wenn nicht aktiviert, dann zeigt Mollie diese Felder direkt auf der externen Mollie Zahlungsseite.</helpText>
             <helpText lang="nl-NL">Toon credit card invoervelden direct in jouw Shopware shop. Indien uitgeschakeld, zal Mollie de formulieren tonen op de Mollie betaalpagina.</helpText>
         </input-field>
+        <input-field type="bool">
+            <name>enableSingleClickPayments</name>
```
Comment (boxblinkracer) [inline src/Resources/config/config.xml:81]: please use new wording + invert the topic...then we have a prefix for features

"oneClickPaymentsEnabled"

Comment (boxblinkracer) [reply src/Resources/config/config.xml:81]: also in the php config class of course

File: src/Resources/config/config.xml [hunk 4]
```diff
@@ -77,6 +77,26 @@
...
+            <label>Enable Single Click Payments</label>
+            <label lang="de-DE">Single-Click-Zahlungen aktivieren</label>
+            <label lang="nl-NL">Betalingen met één klik mogelijk maken</label>
+            <defaultValue>true</defaultValue>
+            <helpText>Save customer's credit card for recurring payments.</helpText>
+            <helpText lang="de-DE">Speichern Sie die Kreditkarte des Kunden für wiederkehrende Zahlungen.</helpText>
+            <helpText lang="nl-NL">Sla de creditcard van de klant op voor terugkerende betalingen.</helpText>
+        </input-field>
+        <input-field type="bool">
+            <name>enableCreditCardCompactView</name>
```
Comment (boxblinkracer) [inline src/Resources/config/config.xml:91]: also here "oneClickPaymentsCompactView"
and please write at least in the label "One-Click Payments compact view"
i will do the rest when it comes to maybe moving the stuff around...(if necessary)

File: src/Resources/views/mollie/component/credit-card-mandate-remove-modal.html.twig [hunk 5]
```diff
@@ -0,0 +1,72 @@
+{% block credit_card_mandate_remove_modal %}
```
Comment (boxblinkracer) [inline src/Resources/views/mollie/component/credit-card-mandate-remove-modal.html.twig:1]: i think all our blocks have a "mollie" prefix in the name

File: src/Subscriber/AccountPaymentMethodPageSubscriber.php [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+    /**
+     * Adds the components variable to the storefront.
+     *
+     * @param AccountPaymentMethodPageLoadedEvent $args
+     */
+    private function addMollieSingleClickPaymentDataToPage(AccountPaymentMethodPageLoadedEvent $args): void
+    {
+        $args->getPage()->assign([
+            'enable_single_click_payments' => $this->settings->isEnableSingleClickPayments(),
```
Comment (boxblinkracer) [inline src/Subscriber/AccountPaymentMethodPageSubscriber.php:75]: could you add a mollie prefix...i think we should go for that (also in twig)

**Review by BlackScorp — APPROVED** [review]
approve

**Pull request merged**